### PR TITLE
Some prerequisites for using IndexedDB to cache data by JSON key.

### PR DESCRIPTION
**Pull request by alxhub**
Addresses #8.

### CODE REVIEW

File: modules/tactical/src/idb.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+/// <reference path="../../../typings/rx/rx-lite.d.ts" />
+
+
+import {Observable} from 'rx';
+
+/**
+ * A very minimal interface for a key-value store, designed as an abstraction
+ * layer over IndexedDB.
+ */
+export interface Idb {
```
Comment (ttowncompiled) [inline modules/tactical/src/idb.ts]: If in the future we would want to allow db adapters, should this interface be named something less specific to indexDB?

File: modules/tactical/test/idb.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/// <reference path="../../../typings/chai/chai.d.ts" />
```
Comment (ttowncompiled) [inline modules/tactical/test/idb.ts]: It is angular convention to name tests, <name>_spec.ts -> s/idb.ts/idb_spec.ts

File: modules/tactical/test/idb.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    });
+  });
+  it('returns undefined for non-existent keys', (done) {
+    var idb = new InMemoryIdb();
+    idb.get('x', 'not-there').subscribe((v) => {
+      expect(v).to.be.undefined;
+      done();
+    });
+  });
+});
```
Comment (ttowncompiled) [inline modules/tactical/test/idb.ts]: Do we care if there is no newline at EOF?

File: modules/tactical/test/json.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    expect(serializeValue(null)).to.equal('null');
+  });
+  it('properly serializes an array', () => {
+    expect(serializeValue([1, 2, 3])).to.equal('[1,2,3]');
+  });
+  it('properly serializes an object', () => {
+    expect(serializeValue({foo: 'bar', baz: 'qux'})).to.equal(
+        '{"baz":"qux","foo":"bar"}');
+  });
+});
```
Comment (ttowncompiled) [inline modules/tactical/test/json.ts]: Same question about EOF here.

File: modules/tactical/test/json.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/// <reference path="../../../typings/chai/chai.d.ts" />
```
Comment (ttowncompiled) [inline modules/tactical/test/json.ts]: s/json/json_spec

See comment above!

**Comment by ttowncompiled** [comment]
Did you run gulp check-format?

File: modules/tactical/test/idb.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+          expect(v3['bar']).to.equal('baz');
+          idb.get('y', 'foo').subscribe((v4) => {
+            expect(v4['bar']).to.equal('qux');
+            done();
+          });
+        });
+      });
+    });
+  });
+  it('returns undefined for non-existent keys', (done) {
```
Comment (ttowncompiled) [inline modules/tactical/test/idb.ts]: forgot fat arrow. Should be
(done) => {

File: modules/tactical/src/idb.ts (hunk 1 above)
Comment (alxhub) [inline modules/tactical/src/idb.ts]: We can rename it later - it's just internal.

File: modules/tactical/test/idb.ts (hunk 2 above)
Comment (alxhub) [inline modules/tactical/test/idb.ts]: Done.

File: modules/tactical/test/json.ts (hunk 4 above)
Comment (alxhub) [inline modules/tactical/test/json.ts]: Fixed.

File: modules/tactical/test/json.ts (hunk 5 above)
Comment (alxhub) [inline modules/tactical/test/json.ts]: Fixed.

File: modules/tactical/test/idb.ts (hunk 6 above)
Comment (alxhub) [inline modules/tactical/test/idb.ts]: Fixed.

**Comment by alxhub** [comment]
Also ran gulp check-format.

**Comment by ttowncompiled** [comment]
LGTM

**Pull request merged**